components/arm/collectors_test.go

```diff
@@ -0,0 +1,118 @@
+package arm
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	clk "github.com/benbjohnson/clock"
+	"github.com/edaniels/golog"
+	"github.com/golang/geo/r3"
+	v1 "go.viam.com/api/common/v1"
+	pb "go.viam.com/api/component/arm/v1"
+	"go.viam.com/test"
+	"go.viam.com/utils/protoutils"
+
+	"go.viam.com/rdk/data"
+	"go.viam.com/rdk/resource"
+	"go.viam.com/rdk/spatialmath"
+	tu "go.viam.com/rdk/testutils"
+)
+
+type collectorFunc func(resource interface{}, params data.CollectorParams) (data.Collector, error)
```

Comment:
Nit/opinion: I think collectorConstructor would be a bit more accurate here. At first read, I assumed this was the capture function

Reply:
Looks like this is a copy of `data.CollectorConstructor` actually! I think you can just use that